### PR TITLE
Help: move search query to local state, remove from Redux

### PR DESCRIPTION
**Pull request by jsnajdr**
Removes the `state.inlineHelp.searchQuery` Redux state and uses local `useState` in respective components instead.

The `InlineHelpSearchCard` component, which shows the search input, no longer dispatches the `setInlineHelpSearchQuery` action but calls the `onSearch` callback prop instead. The parent component then wire up `onSearch` to set the local state. And the search card's `query` prop is being renamed to `searchQuery`, for consistency with other components.

The `InlineHelpRichResult` component no longer gets `searchQuery` from Redux, but gets it from the parent component (`InlineHelpPopover`) which already knows it.

The three search components that use `searchQuery`: inline help popover search, my home search, and masterbar popup search, all now have `searchQuery` in local state and pass it down to the components that need it, mostly just the `InlineHelpSearchCard`.

### CODE REVIEW

File: client/blocks/inline-help/popover.jsx [hunk 1]
```diff
@@ -121,13 +121,14 @@ class InlineHelpPopover extends Component {
 				<QuerySupportTypes />
 				<div className="inline-help__search">
 					<InlineHelpSearchCard
-						query={ this.props.searchQuery }
+						searchQuery={ this.state.searchQuery }
+						onSearch={ ( searchQuery ) => this.setState( { searchQuery } ) }
```
Comment (tyxla) [inline client/blocks/inline-help/popover.jsx]: Should we declare this method separately in a class component, or it's fine if we redeclare it on re-render?

File: client/my-sites/customer-home/cards/features/help-search/index.jsx [hunk 2]
```diff
@@ -38,7 +40,7 @@ const HelpSearch = ( { searchQuery, track } ) => {
 			} ).filter( ( [ , value ] ) => typeof value !== 'undefined' )
 		);
 
-		track( `calypso_inlinehelp_${ type }_open`, tracksData );
+		dispatch( recordTracksEvent( `calypso_inlinehelp_${ type }_open`, tracksData ) );
```
Comment (tyxla) [inline client/my-sites/customer-home/cards/features/help-search/index.jsx:43]: Ah. It's a bummer we generate this event name dynamically. This is totally against the naming conventions as per PCYsg-4S2-p2:

> Don’t programmatically create event names. They should be hardcoded so they are easily searchable in the code base.

Don't feel obligated to change that in this PR though. This is just an observation.

**Review by tyxla — APPROVED** [review]
This works well for me and code looks good 🚢 

Note that I've deleted another unused selector - `hasInlineHelpAPIResults`. It was using the reducer tree that we're removing in this PR.

File: client/my-sites/customer-home/cards/features/help-search/index.jsx (hunk 2 above)
Comment (jsnajdr) [reply client/my-sites/customer-home/cards/features/help-search/index.jsx:43]: Unfortunately, dynamic event names like this are quite common 🙁

File: client/blocks/inline-help/popover.jsx (hunk 1 above)
Comment (jsnajdr) [reply client/blocks/inline-help/popover.jsx]: That's a good idea, pushed a commit with a `setSearchQuery` method :+1:

File: client/my-sites/customer-home/cards/features/help-search/index.jsx (hunk 2 above)
Comment (tyxla) [reply client/my-sites/customer-home/cards/features/help-search/index.jsx:43]: 🤮 🤷‍♂️

**Comment by worldomonation** [comment]
@jsnajdr Did this change alter the `Use a domain I own` screen?

We started seeing permanent failures of the E2E tests on both desktop and mobile around 0800 UTC-8 today (October 29).

**Comment by jsnajdr** [comment]
> Did this change alter the Use a domain I own screen?

@worldomonation That sounds unlikely: this change should affect only the Help popover (after it's opened) and the My Home screen.

There have been quite a few `trunk` commits since you asked on Friday, and the e2e tests are either succeeding, or failing randomly on tests other than "Use a domain I own".

**Comment by worldomonation** [comment]
> > Did this change alter the Use a domain I own screen?
> 
> @worldomonation That sounds unlikely: this change should affect only the Help popover (after it's opened) and the My Home screen.

Okay, thanks. I thought to ask because the build failure started from the merge of this change into `trunk`.

> There have been quite a few `trunk` commits since you asked on Friday, and the e2e tests are either succeeding, or failing randomly on tests other than "Use a domain I own".

Yes, the specific test in question is no longer failing since I've disabled the test pending investigation..

**Pull request merged**